### DISCUSSION TITLE
crossgen2

### DISCUSSION THREAD

**Discussion by snikeguo** [nanoframework/Home#1124] [General]
Coreclr + crossgen2 compilation (most of the code runs directly in flash through XIP mode, and only the emit part runs in RAM) + a little bit of bcl basic library.

The clipped coreclr + BCL libs occupies about 512KB at most, and a “helloword ” project needs about 100kb of ram. All C # features can be used and are fast.

I'm studying coreclr

**Comment by josesimoes** [comment]
Interesting. Care to join our Discord server? It's a better place to keep a conversation.

**Reply by snikeguo** [reply]
It's a pity that China has GFW network censorship, so I can't log in to Discord
